github/github.go

```diff
@@ -392,21 +392,25 @@ func (client *Client) PullRequestForSHA(repo *github.Repository, SHA string) (*g
 		return nil, nil
 	}
 
+	if pullRequest.Mergeable != nil {
+		return pullRequest, nil
+	}
+
 	ticker := time.NewTicker(5 * time.Second)
 	for {
+		select {
+		case <-ticker.C:
+		}
+
 		// polling for merge status (https://developer.github.com/v3/pulls/#get-a-single-pull-request)
 		p, _, err := client.PullRequests.Get(client.Ctx, owner, name, pullRequest.GetNumber())
 		if err != nil {
 			return nil, errors.Wrapf(err, "cannot poll for PR merge status with head %s on %s", SHA, repo.GetFullName())
 		}
-		if p.GetMergeable() {
+		if p.Mergeable != nil {
```

Comment:
Now return the pull request whether it is mergeable or not, whereas before,we only returned it if was mergeable. Do we check the mergeability of the PR again in one of the callers of this function before taking action? If not, do we even need to wait for `p.Mergeable` to be non-nil?

Reply:
It is checked at https://github.com/palantir/bulldozer/blob/master/server/endpoints/hook.go#L202.